bdk-ffi/src/lib.rs

```diff
@@ -1226,12 +1250,34 @@ mod test {
     #[test]
     fn test_extend_descriptor_keys() {
         let master_dsk = get_descriptor_secret_key();
-        let extended_dsk: &DescriptorSecretKey = &extend_dsk(&master_dsk, "m/0");
+        let extended_dsk: &DescriptorSecretKey = &extend_dsk(&master_dsk, "m/0").unwrap();
         assert_eq!(extended_dsk.as_string(), "tprv8ZgxMBicQKsPdWuqM1t1CDRvQtQuBPyfL6GbhQwtxDKgUAVPbxmj71pRA8raTqLrec5LyTs5TqCxdABcZr77bt2KyWA5bizJHnC4g4ysm4h/0/*");
 
         let master_dpk: &DescriptorPublicKey = &master_dsk.as_public();
-        let extended_dpk: &DescriptorPublicKey = &extend_dpk(master_dpk, "m/0");
+        let extended_dpk: &DescriptorPublicKey = &extend_dpk(master_dpk, "m/0").unwrap();
         assert_eq!(extended_dpk.as_string(), "tpubD6NzVbkrYhZ4WywdEfYbbd62yuvqLjAZuPsNyvzCNV85JekAEMbKHWSHLF9h3j45SxewXDcLv328B1SEZrxg4iwGfmdt1pDFjZiTkGiFqGa/0/*");
+
+        let wif = "L2wTu6hQrnDMiFNWA5na6jB12ErGQqtXwqpSL7aWquJaZG8Ai3ch";
+        let extended_key = DescriptorSecretKey::from_string(wif.to_string()).unwrap();
+        let result = extended_key.derive(Arc::new(DerivationPath::new("m/0".to_string()).unwrap()));
+        dbg!(&result);
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_from_str_descriptor_secret_key() {
+        let key1 = "L2wTu6hQrnDMiFNWA5na6jB12ErGQqtXwqpSL7aWquJaZG8Ai3ch";
+        let key2 = "tprv8ZgxMBicQKsPcwcD4gSnMti126ZiETsuX7qwrtMypr6FBwAP65puFn4v6c3jrN9VwtMRMph6nyT63NrfUL4C3nBzPcduzVSuHD7zbX2JKVc/1/1/1/*";
+
+        let private_descriptor_key1 = DescriptorSecretKey::from_string(key1.to_string()).unwrap();
```

Comment:
would it work to initialize with a string with a fingerprint as well?

Reply:
Yep. For example you can add this test and it will compile:
```rust
#[test]
fn test_from_str_descriptor_public_key() {
    let key = "[78412e3a/44'/0'/0']xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL";
    let public_descriptor = DescriptorPublicKey::from_string(key.to_string()).unwrap();
    dbg!(public_descriptor.as_string());
}
```

Reply:
Awesome thanks 👍